test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -1464,6 +1464,62 @@ public static IEnumerable<object[]> OptionalParamValues
         }
 
         [Fact]
+        public async Task AttributeRoute_ReplacesExistingRouteValues_IfNotNull()
```

Comment:
It will retain the default value(if exists) or will be set as null. It won't be set as empty string.